user-console/src/main/resources/org/pentaho/mantle/public/Mantle.jsp

```diff
@@ -207,6 +207,7 @@
     <div id="pucHeader" class="responsive" cellspacing="0" cellpadding="0">
       <div id="pucMenuBar"></div>
       <div id="pucBurgerToolbar"></div>
+      <div id="pucTabsMenuBar" class="flex-row"></div>
```

Comment:
why is the class `"flex-row" `aded here in Mantle.jsp. Can it be handled in the css.

Reply:
So, in this case, and the one referenced (where I was wrong), you cannot really add the class to a CSS stylesheet. 

Unfortunately, CSS does not allow to assign a classes to a selector.. — would be a kind of inheritance, it's a long time ask to CSS, but a wish never fulfilled.

My confusion came from, in some cases, I've been asking people to not assign the CSS class to either HTML/XUL, as shown, or programmatically via Java/JS code, but to instead specify the associated/corresponding CSS properties in a CSS style-sheet.

It's an attempt to follow the principle many defend of keeping style rules in style sheets, and out of markup, but it's achieved at the cost of CSS-layout logic duplication (because there is no inheritance as explained above). Those who defend this approach consider the opposite to be a CSS anti-pattern.

Also, note that this only applies to CSS classes which represent a certain layout or style (e.g. "flex-column", "flex-wrap"). In this approach, CSS classes would instead stand for "classes of things" (e.g. menu, sidebar) or "modifiers of things" thereof (e.g. "critical").

All and all, since we've introduced these CSS classes standing for actual styling and layout, it's kind of a lost battle, and convenience wins.

Reply:
I thought you had added [these reusable classes](https://github.com/pentaho/pentaho-commons-gwt-modules/blob/3b4491b6786b01963742fb8b1f30845203ddcf60/widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css#L999) so that we were not writing the same styles all over the place and instead reusing the classes?

I am fine with whatever convention, I just want to have a firm understanding of the concepts and how to proceed. I am still relatively new to this stuff, I am mostly following the patterns and guidance of others.

Reply:
i'm fine either way, with without the change.